backend/collab_service/controllers/collabController.js

```diff
@@ -46,6 +46,13 @@ class CollabController {
       });
     });
 
+    socket.on('languageUpdate', (data) => {
+      const { sessionIdObj, language } = data;
+      console.log(`Received language update request for session: ${sessionIdObj}`);
+      socket.to(sessionIdObj).emit('languageUpdated', { language });
```

Comment:
Great usage of `socket.to` instead of `io.to` 💯

Reply:
credit to @qinxutan ! 😆